Prevents creation of custom registration form without DisplayName and no DisplayNameFormat

tauqeer-haider opened the pull request:
DNN-41046: disallowed custom form without DisplayName and lacking a DisplayNameFormat

<!-- 
  Please read contribution guideline first: https://github.com/dnnsoftware/Dnn.Platform/blob/develop/CONTRIBUTING.md 
-->

<!-- 
  Please make sure that there is a corresponding issue created and reference it in the PR by writing
  `Fixes #123` or `Closes #123`. 
  A PR without an accompanying issue will be accepted and merged on a very rare occasion
-->
Fixes #3971 

## Summary
<!-- 
  Please describe the code changes as you see fit so that the reviewers have an easier task understanding what changed and why.
  New unit tests will be highly appreciated.
-->
Custom Registration Form without DisplayName field on it and lacking Display Name Format is no longer allowed and considered as incorrect configuration.

valadas commented:
@tauqeer-haider  Please do not use external reference number in PR or issue titles as we use titles for the release notes and those references are internal and mean nothing to visitors here. I updated the title and moved your reference number in the description. Thanks.

tauqeer-haider commented:
> @tauqeer-haider Please do not use external reference number in PR or issue titles as we use titles for the release notes and those references are internal and mean nothing to visitors here. I updated the title and moved your reference number in the description. Thanks.

Noted. Thanks.
